### PR TITLE
Add jquery-spellchecker w/ git auto-update via single package.json

### LINKED ISSUE #11511: [Request] Add jquery-spellchecker
**Library name:** jquery-spellchecker
**Git repository url:** https://github.com/badsyntax/jquery-spellchecker
**npm package name or url** (if there is one): 
**License:** MIT
**Official homepage:** http://jquery-spellchecker.badsyntax.co/
**Wanna say something? Leave message here:**

The files we needs should be the `js` and `css` folders under `dist`

=====================

For beginners who is interested in picking up this issue:

In this issue, we are going to host a new web front-end library on CDNJS, so that web developer can easily use it on our free CDN without additional download/upload process 🚀

If you'll like to work locally, I'll suggest you start pulling cdnjs repository with sparseCheckout: https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md, since there are so many files in this repo, without sparseCheckout, the whole process will be slow and will use more than 80GB disk space, using git command line can learn more skills and experience, if you don't like that, instead, you can also just do it on GitHub, it's more convenient, but it may be a little bit harder to fix problems on GitHub gui.

Here are the docs about how to add a library:
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#d-adding-a-new-library-with-its-assets
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson

We'll also ask to setup auto-update config so that the added libraries can be updated to the latest version easily:
 - https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

For beginners who is interested in join and help add this library, please feel free to tell me what kind of help you'll need, we'll help you send the valid pull request, thanks 😄

### PR DESCRIPTION
**Pull request by sammorrow**
close #11511
Pull request for issue: #11511 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/badsyntax/jquery-spellchecker
 * Official website (optional, not the repository):  http://jquery-spellchecker.badsyntax.co/
 * NPM package url (optional):
 * License and its reference: "MIT" - https://opensource.org/licenses/MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 30
   - Count of stars: 262
   - Count of forks: 108
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sammorrow** [comment]
I cannot resolve this final error: "jquery-spellchecker: filename from package.json exists and is a file (not a directory)"--please help.

File: ajax/libs/jquery-spellchecker/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "url": "https://github.com/badsyntax/jquery-spellchecker"
+  },
+  "keywords": [
+    "jquery",
+    "spellcheck"
+  ],
+  "version": "0.2.4-ignored",
+  "npmName": "jquery-spellchecker",
+  "npmFileMap": [
+    {
```
Comment (sufuf3) [inline ajax/libs/jquery-spellchecker/package.json]: The library only from GitHub, not npm.
So we use git auto-update. Let's modify from A to B.

- A
```js
  "npmName": "jquery-spellchecker",
  "npmFileMap": [
    {
      "basePath": "dist/js",
      "files": [
        "**/*"
      ]
    }
  ]
```

- B
```js
  "autoupdate": {
    "source": "git",
    "target": "git://github.com/badsyntax/jquery-spellchecker.git",
    "fileMap": [
      {
        "basePath": "dist",
        "files": [
          "+(js|css)/jquery.spellchecker*"
        ]
      }
    ]
  }
```

Thank you.

**Comment by sammorrow** [comment]
Thank you for the suggestions @sufuf3 !

**Comment by sufuf3** [comment]
@sammorrow May I ask you how do you send this pull request? via GitHub website or use git command?
I think the commit should squash to one commit, if you use git command, we can help you how to do this.
But if you use GitHub website, it can't squash the commit. I can help you squash to one commit.

BTW, the commit message needs to change as the followings:
```
Add jquery-spellchecker w/ git auto-update via single package.json

close #11511, cc @badsyntax
```

Thank you.

**Comment by sufuf3** [comment]
And could you fill in `Profile of the lib` in the first comment?
It will help reviewers know more about this PR info.
Thank you. 😃

**Comment by sammorrow** [comment]
I used the Github website to fill this out.

**Comment by sufuf3** [comment]
Help squashed the commit and rebased to the master.
@cdnjs/active-contributor @cdnjs/intern3 
Please help review this PR, thank you.

File: ajax/libs/jquery-spellchecker/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "jquery-spellchecker",
+  "filename": "jquery.spellchecker.min.js",
```
Comment (PeterDaveHello) [inline ajax/libs/jquery-spellchecker/package.json]: @sufuf3 @extend1994 the filename needs to contain `js/` here.

Comment (sufuf3) [inline ajax/libs/jquery-spellchecker/package.json]: Yes, Sorry I'll pay attention next time.

Comment (extend1994) [inline ajax/libs/jquery-spellchecker/package.json]: Sorry about carelessness...

**Pull request merged**